controllers/chat/ChatController.js

```diff
@@ -156,13 +156,13 @@ module.exports = {
     }
   },
   getChannels: async (req, res) => {
-    const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    const client = new StreamChat(process.env.API_KEY, process.env.SECRET);
     try {
       await client.connectUser({id: req.userId}, req.token);
       let channels = await client.queryChannels({members: {$in: [req.userId]}}, [
         {last_message_at: -1}
       ]);
-      await client.disconnectUser();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The creation of a new `StreamChat` instance every time the `getChannels` function is called could lead to performance issues. If the `StreamChat` class has expensive setup operations, creating a new instance for each request can slow down your application. Consider using a singleton pattern or connection pooling to manage `StreamChat` instances.

```diff
- const client = new StreamChat(process.env.API_KEY, process.env.SECRET);
+ const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->